File: Frontend/services/project.service.js
import {
    get,
    post,
    put,
    remove
}
from "./api.js";

export async function getProjects(){

    return await get(
        "/projects"
    );
}

export async function getProject(
    projectId
){

    return await get(
        `/projects/${projectId}`
    );
}

export async function createProject(
    payload
){

    return await post(
        "/projects",
        payload
    );
}

export async function updateProject(
    projectId,
    payload
){

    return await put(
        `/projects/${projectId}`,
        payload
    );
}

export async function deleteProject(
    projectId
){

    return await remove(
        `/projects/${projectId}`
    );
}

File: Backend/services/project_service.py
from sqlalchemy.orm import Session
from models.project import Project
from schemas.project import ProjectCreate

class ProjectService:

    def create_project(self, db: Session,payload: ProjectCreate) -> Project:
        project = Project(name=payload.name,description=payload.description)
        
        db.add(project)
        db.commit()
        db.refresh(project)

        return project

    def get_projects(self, db: Session) -> list[Project]:
        return (db.query(Project).order_by(Project.updated_at.desc()).all())

    def get_project_by_id(self, db: Session,project_id: int) -> Project | None:
        return (db.query(Project).filter(Project.id == project_id).first())

    def delete_project(self, db: Session, project_id: int) -> bool:
        project = (db.query(Project).filter(Project.id == project_id).first())

        if not project:
            return False

        db.delete(project)

        db.commit()

        return True